Clear/EVMState.lean

```diff
@@ -163,6 +164,60 @@ instance : Inhabited EVMState :=
 
 abbrev EVM := EVMState
 
+instance instFinmapPreorder : Preorder (Finmap (λ _ : List UInt256 ↦ UInt256)) where
+  le a b := a.keys ⊆ b.keys ∧ ∀ {key}, key ∈ a → a.lookup key = b.lookup key
+  le_refl := by aesop
+  le_trans := by aesop (add safe forward subset_trans)
+
+def preserved : EVMState → EVMState → Prop :=
+  λ a b ↦ (Eq on EVMState.account_map) a b ∧
+          (Eq on EVMState.hash_collision) a b ∧
+          (Eq on EVMState.execution_env) a b ∧
+          (a.keccak_map ≤ b.keccak_map)
+
+lemma preserved_def {σ₀ σ₁ : EVM} : preserved σ₀ σ₁ =
+  (σ₀.account_map    = σ₁.account_map ∧
+   σ₀.hash_collision = σ₁.hash_collision ∧
+   σ₀.execution_env  = σ₁.execution_env ∧
+   σ₀.keccak_map     ≤ σ₁.keccak_map) := by
+  unfold preserved
+  dsimp
+
+def preserved_account_map {evm evm' : EVMState} (h : preserved evm evm') :
+  evm.account_map = evm'.account_map := by
+  rw [preserved_def] at h
+  exact h.1
+
+def preserved_collision {evm evm' : EVMState} (h : preserved evm evm') :
+  evm.hash_collision = evm'.hash_collision := by
+  rw [preserved_def] at h
+  exact h.2.1
+
+def preserved_execution_env {evm evm' : EVMState} (h : preserved evm evm') :
+  evm.execution_env = evm'.execution_env := by
+  rw [preserved_def] at h
+  exact h.2.2.1
+
+def mono_keccak_map {evm evm' : EVMState} (h : preserved evm evm') :
```

Comment:
This is just a projection function, it's not a monotonicity statement. It says that you can take out the last component of and.